### PR TITLE
feat: Implement dynamic left and right prompt prefix options

### LINKED ISSUE #148: Timestamp on prompt
Hi,

Is it possible to append a timestamp on the prompt? It would be helpful to know when a command was run.

I've tried with `TYPEWRITTEN_RIGHT_PROMPT_PREFIX`, but it seems it is only read on start, so `date` doesn't work :(

### PR DESCRIPTION
**Pull request by reobin**
Closes #148 

4 "new" options were implemented:

- `TYPEWRITTEN_LEFT_PROMPT_PREFIX` (existed before)
- `TYPEWRITTEN_LEFT_PROMPT_PREFIX_FUNCTION`
- `TYPEWRITTEN_RIGHT_PROMPT_PREFIX`
- `TYPEWRITTEN_RIGHT_PROMPT_PREFIX_FUNCTION`

The `_FUNCTION` variants are used to enable the use of live evaluated functions as prefixes.

Example of use in `.zshrc` to display kube context:

```sh
display_kube_context() {
  tw_kube_context="$(kubectl config current-context 2> /dev/null)"

  if [[ $tw_kube_context != "" ]]; then
    echo "($(basename $tw_kube_context))"
  fi
}

export TYPEWRITTEN_LEFT_PROMPT_PREFIX_FUNCTION=display_kube_context
```

### CODE REVIEW

**Comment by reobin** [comment]
Hello @KEDLogic ! If you have the chance to review this PR, it would be greatly appreciated. Especially the example snippet for kube context.

**Comment by KEDLogic** [comment]
Oh, very nice! I'll carve out some time tomorrow to go over this. 

In case I somehow miss it when I review, the `2&>1` in that snippit needs to changed to `2>&1`. 

That bug of mine caused a fun little hunt for the source of all the `1` files throughout my file system...

**Comment by KEDLogic** [comment]
I gave it a quick test and it looks good! 

That snippet needs a small edit for the error redirect (My original script had an error): 
```
display_kube_context() {
  tw_kube_context="$(kubectl config current-context 2> /dev/null)"

  if [[ $tw_kube_context != "" ]]; then
    echo "($(basename $tw_kube_context))"
  fi
}

export TYPEWRITTEN_LEFT_PROMPT_PREFIX_FUNCTION=display_kube_context
```

File: docs/prompt_customization.md [hunk 1]
```diff
@@ -167,10 +170,77 @@ This option being off means that at all times, the current directory display onl
...
+  <img src="_media/left_prompt_prefix/custom_command.png" alt="bash prompt date prefix" width="750" />
+</p>
+
+**Bonus example**
+
+Display kube_context:
+
+```sh
+display_kube_context() {
+  tw_kube_context="$(kubectl config current-context 2&>1 /dev/null)"
```
Comment (KEDLogic) [inline docs/prompt_customization.md:238]: The error redirect should be: 
```
  tw_kube_context="$(kubectl config current-context 2> /dev/null)"
```

**Review by KEDLogic — CHANGES_REQUESTED** [review]
Looks good! 
It just needs a small edit to the kubecontext snippet as noted.

**Review by KEDLogic — APPROVED** [review]

**Comment by reobin** [comment]
@all-contributors please add @KEDLogic for documentation and ideas

**Pull request merged**